### PR TITLE
Refactor node annotations

### PR DESCRIPTION
**Pull request by marseel**
Refactor node annotations and add function for removing node's annotations.

Related: #22306

### CODE REVIEW

File: daemon/cmd/daemon.go [hunk 1]
```diff
@@ -1182,7 +1182,7 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup,
 			logfields.V6CiliumHostIP: node.GetIPv6Router(),
 		}).Info("Annotating k8s node")
 
-		err := k8s.AnnotateNode(
+		annotation, err := k8s.AnnotateNode(
```
Comment (squeed) [inline daemon/cmd/daemon.go]: This function is getting out of hand; can we replace most of the parameters with the `node` object?

File: daemon/cmd/daemon.go [hunk 2]
```diff
@@ -1192,7 +1192,15 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup,
 			node.GetInternalIPv4Router(), node.GetIPv6Router())
 		if err != nil {
 			log.WithError(err).Warning("Cannot annotate k8s node with CIDR range")
+		} else {
+			cleaner.cleanupFuncs.Add(func() {
```
Comment (squeed) [inline daemon/cmd/daemon.go]: We should only do this if, as mentioned in the original issue, clean-cilium-state is set.

**Review by squeed — CHANGES_REQUESTED** [review]

File: daemon/cmd/daemon.go (hunk 1 above)
Comment (marseel) [reply daemon/cmd/daemon.go]: Done.

File: daemon/cmd/daemon.go (hunk 2 above)
Comment (marseel) [reply daemon/cmd/daemon.go]: For now, I removed this part and only pushed refactoring + new function for removing node annotations. I will work on this in separate PR.

File: pkg/k8s/annotate_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package k8s
+
+import (
+	"github.com/cilium/cilium/pkg/annotation"
```
Comment (thorn3r) [inline pkg/k8s/annotate_test.go]: got a linting error here. i think this should fix it:
```
goimports -w -local github.com/cilium/cilium/pkg/annotation pkg/k8s/annotate_test.go
```

**Review by thorn3r — APPROVED** [review]
LGTM! 
CI hit a flake tracked here: https://github.com/cilium/cilium/issues/23845

**Review by squeed — APPROVED** [review]

**Comment by squeed** [comment]
Agreed; that seems to be a known flake. Marking as ready-to-merge.

**Pull request merged**